### PR TITLE
add more stm32fx target boards for testing uart_async_api with dma

### PR DESCRIPTION
**Pull request by FRASTM**
This patch is enabling tests/drivers/uart/uart_async_api on the following boards from STMicroelectronics:
nucleo_f207zg
nucleo_f429zi
~~nucleo_f746zg~~
stm32f3_disco

On the nucleo 144 board, the **usart6** is selected for the testcase with DMA2 enabled
Pin  PG14 <--> PG9 (pin 14 & 16 of the connector 10) must be wired to Pass the test

On the disco board, the **usart2** is selected for the testcase with DMA1 enabled
Pin  PA2 <--> PA3 (pin 11 & 14 of the P1 connector ) must be wired to Pass the test

### CODE REVIEW

**Comment by FRASTM** [comment]
running tests/drivers/uart/uart_async_api:

```
*** Booting Zephyr OS build zephyr-v2.6.0-2303-gc8e0f8f7b58d  ***               
Running test suite uart_async_test                                              
===================================================================             
START - test_single_read_setup                                                  
 PASS - test_single_read_setup in 0.1 seconds                                   
===================================================================             
START - test_single_read                                                        
 PASS - test_single_read in 0.402 seconds                                       
===================================================================             
START - test_chained_read_setup                                                 
 PASS - test_chained_read_setup in 0.1 seconds                                  
===================================================================             
START - test_chained_read                                                       
 PASS - test_chained_read in 0.267 seconds                                      
===================================================================             
START - test_double_buffer_setup                                                
 PASS - test_double_buffer_setup in 0.1 seconds                                 
===================================================================             
START - test_double_buffer                                                      
 PASS - test_double_buffer in 3.404 seconds                                     
===================================================================             
START - test_read_abort_setup                                                   
 PASS - test_read_abort_setup in 0.1 seconds                                    
===================================================================             
START - test_read_abort                                                         
 PASS - test_read_abort in 1.110 seconds                                        
===================================================================             
START - test_chained_write_setup                                                
 PASS - test_chained_write_setup in 0.1 seconds                                 
===================================================================             
START - test_chained_write                                                      
 PASS - test_chained_write in 0.3 seconds                                       
===================================================================             
START - test_long_buffers_setup                                                 
 PASS - test_long_buffers_setup in 0.1 seconds                                  
===================================================================             
START - test_long_buffers                                                       
 PASS - test_long_buffers in 0.152 seconds                                      
===================================================================             
START - test_write_abort_setup                                                  
 PASS - test_write_abort_setup in 0.1 seconds                                   
===================================================================             
START - test_write_abort                                                        
 PASS - test_write_abort in 0.102 seconds                                       
===================================================================             
START - test_forever_timeout_setup                                              
 PASS - test_forever_timeout_setup in 0.1 seconds                               
===================================================================             
START - test_forever_timeout                                                    
 PASS - test_forever_timeout in 3.1 seconds                                     
===================================================================             
Test suite uart_async_test succeeded                                            
===================================================================             
PROJECT EXECUTION SUCCESSFUL        
```

**Comment by FRASTM** [comment]
adding a commit for removing the Dcache on the stm32f7xx device when NOCACHE_MEMORY is configured, this must be the case when when DMA is on.

File: soc/arm/st_stm32/stm32f7/soc.c [hunk 1]
```diff
@@ -35,12 +35,12 @@ static int st_stm32f7_init(const struct device *arg)
 	LL_FLASH_EnableART();
 
 	key = irq_lock();
-
+#ifndef CONFIG_NOCACHE_MEMORY
```
Comment (erwango) [inline soc/arm/st_stm32/stm32f7/soc.c:38]: Not the right solution. Let's remove F7 case from this PR.

**Review by erwango — CHANGES_REQUESTED** [review]

File: soc/arm/st_stm32/stm32f7/soc.c (hunk 1 above)
Comment (FRASTM) [reply soc/arm/st_stm32/stm32f7/soc.c]: removing the nucleo_f746zg

**Comment by FRASTM** [comment]
removing the nucleo_f746zg

**Review by erwango — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Comment by erwango** [comment]
Build kite issue should be fixed by https://github.com/zephyrproject-rtos/zephyr/pull/37884.
Can you rebase ?

**Pull request merged**